src/nwm_routing/src/nwm_routing/__main__.py

```diff
@@ -1189,6 +1189,17 @@ def main_v03(argv):
             route_end_time = time.time()
             task_times['route_time'] += route_end_time - route_start_time
 
+        # create initial conditions for next loop itteration
+        q0 = new_nwm_q0(run_results)
+        waterbodies_df = get_waterbody_water_elevation(waterbodies_df, q0)
```

Comment:
I've moved these lines up so that flow and depth states in `q0` and `waterbodies_df` are updated after every loop. This way, a lite restart file is dropped at the very end of the simulation time domain.